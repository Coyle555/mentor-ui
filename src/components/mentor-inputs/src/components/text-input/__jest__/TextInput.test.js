import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, waitForElement, cleanup, act, getByTestId } from '@testing-library/react';

import TextInput from '../textInput';

afterEach(cleanup);

test('<TextInput /> with no props', () => {
	 const component = renderer.create( <TextInput/> );

	 const tree = component.toJSON();
	 expect(tree).toMatchSnapshot();
});

test('<TextInput /> with a disabled attribute', () => {
 	const component = renderer.create(<TextInput disabled={true} />);

 	const tree = component.toJSON();
 	expect(tree).toMatchSnapshot();
});

test('<TextInput /> with a required attribute', () => {
 	const component = renderer.create(<TextInput required={true} />);

 	const tree = component.toJSON();
 	expect(tree).toMatchSnapshot();
});

test('<TextInput /> with value of null', () => {
 	const component = renderer.create(<TextInput value={null} />);

 	const tree = component.toJSON();
 	expect(tree).toMatchSnapshot();
});

test('<TextInput /> with a stringified value', () => {
 	const component = renderer.create(<TextInput value={4593.3} />);

 	const tree = component.toJSON();
 	expect(tree).toMatchSnapshot();
});

test('<TextInput /> with a string value', () => {
 	const component = renderer.create(<TextInput value="Bill" />);

 	const tree = component.toJSON();
 	expect(tree).toMatchSnapshot();
}); 

test('Text input with custom class', () => {
	const tree = renderer.create(<TextInput className="foo" />).toJSON();

	expect(tree).toMatchSnapshot();
});

test('Text input onChange callback', () => {
	const onChange = jest.fn();
	const { container } = render(<TextInput name="text" onChange={onChange} />);

	fireEvent.change(container.querySelector('input'), { target: { value: 'foo' } });
	expect(onChange).toHaveBeenCalledWith(false, 'foo', 'text');
});

test('Text input onBlur callback', () => {
	const onBlur = jest.fn();
	const { container } = render(<TextInput name="text" onBlur={onBlur} />);

	fireEvent.change(container.querySelector('input'), { target: { value: 'foo' } });
	fireEvent.blur(container.querySelector('input'));
	expect(onBlur).toHaveBeenCalledWith(false, 'foo', 'text');
});

test('Custom validation on text input', () => {
	const onChange = jest.fn();
	const validate = jest.fn(val => val === 'adam' ? false : true);
	const { container } = render(<TextInput name="text" onChange={onChange} validate={validate} />);

	fireEvent.change(container.querySelector('input'), { target: { value: 'adam' } });
	expect(onChange).toHaveBeenCalledWith(true, 'adam', 'text');
});
