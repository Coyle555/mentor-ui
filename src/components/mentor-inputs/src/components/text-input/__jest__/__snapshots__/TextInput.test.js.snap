// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> with a required attribute 1`] = `
<div
  className="mui-mi-input-group"
>
  <span
    className="mui-mi-input-addon"
  >
    <i
      className="fal fa-text"
    />
  </span>
  <input
    autoComplete="false"
    className="mui-mi-input-field"
    classes={
      Object {
        "addon": "mui-mi-input-addon",
        "inputGroup": "mui-mi-input-group",
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter text"
    required={true}
    type="text"
    value=""
  />
</div>
`;

exports[`<TextInput /> with a string value 1`] = `
<div
  className="mui-mi-input-group"
>
  <span
    className="mui-mi-input-addon"
  >
    <i
      className="fal fa-text"
    />
  </span>
  <input
    autoComplete="false"
    className="mui-mi-input-field"
    classes={
      Object {
        "addon": "mui-mi-input-addon",
        "inputGroup": "mui-mi-input-group",
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter text"
    type="text"
    value="Bill"
  />
</div>
`;

exports[`<TextInput /> with a stringified value 1`] = `
<div
  className="mui-mi-input-group"
>
  <span
    className="mui-mi-input-addon"
  >
    <i
      className="fal fa-text"
    />
  </span>
  <input
    autoComplete="false"
    className="mui-mi-input-field"
    classes={
      Object {
        "addon": "mui-mi-input-addon",
        "inputGroup": "mui-mi-input-group",
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter text"
    type="text"
    value="4593.3"
  />
</div>
`;

exports[`<TextInput /> with autocomplete enabled 1`] = `
<div
  className="mui-mi-input-group"
>
  <span
    className="mui-mi-input-addon"
  >
    <i
      className="fal fa-text"
    />
  </span>
  <input
    autoComplete="true"
    className="mui-mi-input-field"
    classes={
      Object {
        "addon": "mui-mi-input-addon",
        "inputGroup": "mui-mi-input-group",
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter text"
    type="text"
    value=""
  />
</div>
`;

exports[`<TextInput /> with no props 1`] = `
<div
  className="mui-mi-input-group"
>
  <span
    className="mui-mi-input-addon"
  >
    <i
      className="fal fa-text"
    />
  </span>
  <input
    autoComplete="false"
    className="mui-mi-input-field"
    classes={
      Object {
        "addon": "mui-mi-input-addon",
        "inputGroup": "mui-mi-input-group",
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter text"
    type="text"
    value=""
  />
</div>
`;

exports[`<TextInput /> with value of null 1`] = `
<div
  className="mui-mi-input-group"
>
  <span
    className="mui-mi-input-addon"
  >
    <i
      className="fal fa-text"
    />
  </span>
  <input
    autoComplete="false"
    className="mui-mi-input-field"
    classes={
      Object {
        "addon": "mui-mi-input-addon",
        "inputGroup": "mui-mi-input-group",
      }
    }
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    placeholder="Enter text"
    type="text"
    value=""
  />
</div>
`;
